### PR TITLE
Add output struct to by scs generation

### PR DESCRIPTION
**Pull request by SHtress**
Depends on [PR](https://github.com/ostis-ai/sc-machine/pull/208/files) in sc-machine

### CODE REVIEW

File: src/sc_client/client/_payload_factory.py [hunk 1]
```diff
@@ -71,10 +72,27 @@ def solve_adj(obj: ScAddr | str):
 
 
 class CreateElementsBySCsPayloadCreator(BasePayloadCreator):
-    def __call__(self, scs: SCsText, *_):
-        if not isinstance(scs, get_origin(SCsText)) or not all(isinstance(n, get_args(SCsText)) for n in scs):
-            raise exceptions.InvalidTypeError("expected object types: List[str]")
-        return scs
+    def __call__(self, scs_list: SCsText, *_):
+        if not isinstance(scs_list, list) or not all(isinstance(n, (str, SCs)) for n in scs_list):
```
Comment (AlexandrZagorskiy) [inline src/sc_client/client/_payload_factory.py]:
```suggestion
    def __call__(self, scs_list: List[SCsText], *_):
        if not isinstance(scs_list, list) or not all(isinstance(n, (str, SCs)) for n in scs_list):
```
Update typing, if there is List now

File: src/sc_client/client/_payload_factory.py [hunk 2]
```diff
@@ -71,10 +72,27 @@ def solve_adj(obj: ScAddr | str):
...
+            raise exceptions.InvalidTypeError("expected object types: SCsText")
```
Comment (AlexandrZagorskiy) [inline src/sc_client/client/_payload_factory.py]: If this message will be returned to the user of the API, he wouldn't understand what type is SCsText

File: src/sc_client/constants/common.py [hunk 3]
```diff
@@ -58,6 +58,9 @@ class ClientCommand(Enum):
 
 ELEMENT = "el"
 
+SCS = "scs"
+OUTPUT_STRUCTURE = "outputStructure"
```
Comment (AlexandrZagorskiy) [inline src/sc_client/constants/common.py]: We don't use camel case, rewrite it with underscore

File: src/sc_client/models.py [hunk 4]
```diff
@@ -224,7 +224,13 @@ def get_index(self, alias: str) -> int:
         return self.aliases[alias]
 
 
-SCsText = List[str]
+class SCs:
+    def __init__(self, text: str, output_struct: ScAddr = ScAddr(0)) -> None:
+        self.scs = text
+        self.output_struct = output_struct
```
Comment (AlexandrZagorskiy) [inline src/sc_client/models.py]:
```
class ScTemplateValue(TypedDict):
    value: ScTemplateParamValue
    alias: str | None
```
Make _SCs_ class similar to others

File: tests/api_test.py [hunk 5]
```diff
@@ -121,6 +122,18 @@ def test_create_by_wrong_scs(self):
...
+        self.get_server_message('{"errors": [], "id": 1, "event": false, "status": true, "payload": [1]}')
+        results = client.create_elements_by_scs([SCs("concept1 -> node1;;", ScAddr(0))])
+        assert len(results) == 1
+        assert results[0] is True
+
+    def test_create_by_wrong_scs_with_output_struct(self):
+        self.get_server_message('{"errors": [], "id": 1, "event": false, "status": true, "payload": [0]}')
+        results = client.create_elements_by_scs([SCs("concept1 -> ", ScAddr(0))])
+        assert len(results) == 1
+        assert results[0] is False
```
Comment (AlexandrZagorskiy) [inline tests/api_test.py:135]: message with error description is not returned if error was returned from ScServer? Is it ok? 
There is no tests for passing multiple scs-lines, but format of the request looks like we can do it. Can you add tests for creating multiple scs with multiple output structs> What happend if I will pass a multiple SCs with a common structure?

**Review by AlexandrZagorskiy — CHANGES_REQUESTED** [review]

File: src/sc_client/client/_payload_factory.py (hunk 1 above)
Comment (AlexandrZagorskiy) [reply src/sc_client/client/_payload_factory.py]: Or simply rename _scs_list_ variable to _scs_text_, changing typing is incorrect suggestion

File: src/sc_client/models.py (hunk 4 above)
Comment (SHtress) [reply src/sc_client/models.py]: I change to dataclass cause TypedDict does not support instance and class checks

File: src/sc_client/client/_payload_factory.py [hunk 6]
```diff
@@ -71,10 +72,27 @@ def solve_adj(obj: ScAddr | str):
 
 
 class CreateElementsBySCsPayloadCreator(BasePayloadCreator):
-    def __call__(self, scs: SCsText, *_):
-        if not isinstance(scs, get_origin(SCsText)) or not all(isinstance(n, get_args(SCsText)) for n in scs):
-            raise exceptions.InvalidTypeError("expected object types: List[str]")
-        return scs
+    def __call__(self, scs_text: SCsText, *_):
+        if not isinstance(scs_text, list) or not all(isinstance(n, (str, SCs)) for n in scs_text):
+            raise exceptions.InvalidTypeError("expected object types: string or SCs(string, ScAddr)")
```
Comment (NikitaZotov) [inline src/sc_client/client/_payload_factory.py:77]:
```suggestion
            raise exceptions.InvalidTypeError("Expected object types: string or SCs(string, ScAddr)")
```

File: src/sc_client/client/_payload_factory.py [hunk 7]
```diff
@@ -71,10 +72,27 @@ def solve_adj(obj: ScAddr | str):
...
+                payload.append(
+                    {
+                        common.SCS: scs,
+                        common.OUTPUT_STRUCTURE: 0,
+                    }
+                )
+            else:
+                payload.append(
+                    {
+                        common.SCS: scs.scs,
```
Comment (NikitaZotov) [inline src/sc_client/client/_payload_factory.py]: Replace scs field by text: scs.text

File: docs/changelog.md [hunk 8]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+ - Output struct for generation by scs
```
Comment (NikitaZotov) [inline docs/changelog.md]:
```suggestion
 - Output structure for generation by scs
```

File: src/sc_client/client/_payload_factory.py [hunk 9]
```diff
@@ -71,10 +72,27 @@ def solve_adj(obj: ScAddr | str):
 
 
 class CreateElementsBySCsPayloadCreator(BasePayloadCreator):
-    def __call__(self, scs: SCsText, *_):
-        if not isinstance(scs, get_origin(SCsText)) or not all(isinstance(n, get_args(SCsText)) for n in scs):
-            raise exceptions.InvalidTypeError("expected object types: List[str]")
-        return scs
+    def __call__(self, scs_text: SCsText, *_):
```
Comment (NikitaZotov) [inline src/sc_client/client/_payload_factory.py:75]: Add feature info into readme

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: tests/api_test.py [hunk 10]
```diff
@@ -121,6 +122,18 @@ def test_create_by_wrong_scs(self):
         assert len(results) == 1
         assert results[0] is False
 
+    def test_create_by_scs_with_output_struct(self):
+        self.get_server_message('{"errors": [], "id": 1, "event": false, "status": true, "payload": [1]}')
+        results = client.create_elements_by_scs([SCs("concept1 -> node1;;", ScAddr(0))])
```
Comment (NikitaZotov) [inline tests/api_test.py]: Can we not pass ScAddr(0) as param and use as default in this method?

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: src/sc_client/client/_payload_factory.py (hunk 6 above)
Comment (SHtress) [reply src/sc_client/client/_payload_factory.py:77]: other exceptions start with lowercase

File: src/sc_client/constants/common.py (hunk 3 above)
Comment (NikitaZotov) [reply src/sc_client/constants/common.py]: output_structure

**Review by AlexandrZagorskiy — APPROVED** [review]

File: tests/api_test.py (hunk 5 above)
Comment (SHtress) [reply tests/api_test.py:135]: Add tests for multiple SCs

**Review by NikitaZotov — APPROVED** [review]

**Pull request merged**